storage/innobase/row/row0import.cc

```diff
@@ -3739,12 +3742,13 @@ dberr_t row_import_for_mysql(dict_table_t *table, dd::Table *table_def,
     return (row_import_cleanup(prebuilt, trx, err));
   }
 
-  /*a For encrypted tablespace, set encryption information. */
-  if (FSP_FLAGS_GET_ENCRYPTION(fsp_flags)) {
-    err = fil_set_encryption(
-        table->space,
-        cfg.m_is_keyring_encrypted ? Encryption::KEYRING : Encryption::AES,
-        table->encryption_key, table->encryption_iv);
+  /* For encrypted tablespace, set encryption information.
+  If it is keyring encryption encryption iv was already read in fil_ibd_open
+  and the encryption key will be loaded when keys version is read from pages. */
+
+  if (FSP_FLAGS_GET_ENCRYPTION(fsp_flags) && !cfg.m_is_keyring_encrypted) {
```

Comment:
If it is keyring encrypted, where is the encryption information set?

Reply:
It's in fil_ibd_open - few lines above. For Keyring all needed information is read from page0 - thus it can be assigned in fil_ibd_open. For Master Key the transition key is kept in cfp file - it is used to decrypt the encryption key. It is done from row0import level.